### PR TITLE
Skip initializing the telemetry collector if telemetry is disabled

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

File: pkg/cmd/server/server.go [hunk 1]
```diff
@@ -390,27 +392,32 @@ func (c *Config) Complete(ctx context.Context) (RunnableServer, error) {
...
+	} else if c.TelemetryEndpoint != "" {
+		log.Ctx(ctx).Debug().Msg("initializing telemetry collector")
+		registry, err := telemetry.RegisterTelemetryCollector(c.DatastoreConfig.Engine, ds)
 		if err != nil {
-			return nil, fmt.Errorf("unable to initialize metrics reporter: %w", err)
+			log.Ctx(ctx).Warn().Err(err).Msg("unable to initialize telemetry collector")
+		}
+		telemetryRegistry = registry
+
+		if telemetryRegistry != nil {
```
Comment (ecordell) [inline pkg/cmd/server/server.go:410]: Is this part needed? why not just return on the err on line 405 and then you wont reach the remote reporter init if the registry is nil.

Comment (josephschorr) [reply pkg/cmd/server/server.go:410]: I assumed there were cases where it might be nil even without an error

Comment (ecordell) [reply pkg/cmd/server/server.go:410]: Nope: https://github.com/josephschorr/spicedb/blob/4f89f60bd0ebe154d09ae23a19e055c52180a51d/internal/telemetry/metrics.go#L24-L94

Comment (josephschorr) [reply pkg/cmd/server/server.go]: Okay; updated

**Review by ecordell — APPROVED** [review]
LGTM

**Pull request merged**